packages/discovery-provider/src/api/v1/users.py

```diff
@@ -3007,3 +3011,160 @@ def get(self, id, authed_user_id):
+        params={"id": "A User ID"},
+        responses={
+            200: "Success",
+            400: "Bad request",
+            401: "Unauthorized",
+            403: "Forbidden",
+            500: "Server error",
+        },
+    )
+    @ns.marshal_with(encrypted_email_response)
+    def get(self, id):
+        decoded_id = decode_with_abort(id, ns)
+
+        db = get_db_read_replica()
+        with db.scoped_session() as session:
+            emails = (
+                session.query(EncryptedEmail)
+                .filter(
+                    EncryptedEmail.primary_user_id == decoded_id,
+                )
+                .all()
+            )
+
+            emails = [
+                {
+                    "id": email.id,
+                    "email_owner_user_id": email.email_owner_user_id,
+                    "primary_user_id": email.primary_user_id,
+                    "encrypted_email": email.encrypted_email,
+                    "created_at": email.created_at,
+                    "updated_at": email.updated_at,
+                }
+                for email in emails
+            ]
+
+            return success_response(emails)
+
+
+email_encryption_key_response = make_response(
+    "email_encryption_key_response", ns, fields.Nested(email_encryption_key)
+)
+
+USER_EMAIL_ENCRYPTION_KEY_ROUTE = "/<string:id>/email_encryption_key"
+
+
+@ns.route(USER_EMAIL_ENCRYPTION_KEY_ROUTE)
+class UserEmailEncryptionKey(Resource):
+    @record_metrics
+    @ns.doc(
+        id="""Get Email Encryption Key""",
+        description="""Gets the encryption key for a user with primary access""",
+        params={"id": "A User ID"},
+        responses={
+            200: "Success",
+            400: "Bad request",
+            401: "Unauthorized",
+            403: "Forbidden",
+            500: "Server error",
+        },
+    )
+    @ns.marshal_with(email_encryption_key_response)
+    def get(self, id):
+        decoded_id = decode_with_abort(id, ns)
+
+        db = get_db_read_replica()
+        with db.scoped_session() as session:
+            encryption_key = (
+                session.query(EmailEncryptionKey)
+                .filter(EmailEncryptionKey.primary_user_id == decoded_id)
+                .first()
+            )
+
+            if not encryption_key:
+                abort_not_found(decoded_id, ns)
+
+            encryption_key_dict = {
+                "id": encryption_key.id,
+                "primary_user_id": encryption_key.primary_user_id,
+                "encrypted_key": encryption_key.encrypted_key,
+                "created_at": encryption_key.created_at,
+                "updated_at": encryption_key.updated_at,
+            }
+
+            return success_response(encryption_key_dict)
+
+
+email_access_key_response = make_response(
+    "email_access_key_response", ns, fields.Nested(email_access_key)
+)
+
+USER_EMAIL_ACCESS_KEY_ROUTE = "/<string:id>/email_access_key/<string:primary_user_id>"
```

Comment:
I don't think this route and the other should exist. Include the key in the response for the fan export instead